plugins/modules/vmware_guest_disk.py

```diff
@@ -833,30 +910,49 @@ def gather_disk_facts(vm_obj):
                 if disk.storageIOAllocation is None:
                     disk.storageIOAllocation = vim.StorageResourceManager.IOAllocationInfo()
                     disk.storageIOAllocation.shares = vim.SharesInfo()
+
                 if disk.shares is None:
                     disk.shares = vim.SharesInfo()
-                disks_facts[disk_index] = dict(
-                    key=disk.key,
-                    label=disk.deviceInfo.label,
-                    summary=disk.deviceInfo.summary,
-                    backing_filename=disk.backing.fileName,
-                    backing_datastore=disk.backing.datastore.name,
-                    backing_disk_mode=disk.backing.diskMode,
-                    backing_sharing=disk.backing.sharing,
-                    backing_writethrough=disk.backing.writeThrough,
-                    backing_thinprovisioned=disk.backing.thinProvisioned,
-                    backing_eagerlyscrub=bool(disk.backing.eagerlyScrub),
-                    backing_uuid=disk.backing.uuid,
-                    controller_key=disk.controllerKey,
-                    unit_number=disk.unitNumber,
-                    iolimit_limit=disk.storageIOAllocation.limit,
-                    iolimit_shares_level=disk.storageIOAllocation.shares.level,
-                    iolimit_shares_limit=disk.storageIOAllocation.shares.shares,
-                    shares_level=disk.shares.level,
-                    shares_limit=disk.shares.shares,
-                    capacity_in_kb=disk.capacityInKB,
-                    capacity_in_bytes=disk.capacityInBytes,
-                )
+
+                if isinstance(disk.backing, vim.vm.device.VirtualDisk.RawDiskMappingVer1BackingInfo):
+                    disks_facts[disk_index] = dict(key=disk.key,
+                                                   label=disk.deviceInfo.label,
```

Comment:
Create a dict and add common elements to it and the rest of them in if block.

Reply:
Updated the dict and split out the keys based on whether its a rdm or virtualdisk